Improve constrainable properties sections

anssiko opened the pull request:
Fix #159. PTAL @astojilj


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
[Preview](https://s3.amazonaws.com/pr-preview/w3c/mediacapture-depth/issue-159-constrainable-properties.html) | [Diff](https://s3.amazonaws.com/pr-preview/w3c/mediacapture-depth/7b04774...5fe6430.html)

anssiko committed:
Fix #159: Improve constrainable properties sections

anssiko committed:
Fix depthToVideoTransform definition

anssiko committed:
Align MediaTrackSupportedConstraints per implementation feedback

astojilj commented on index.html:
```diff
@@ -317,7 +321,14 @@
+        The concepts <dfn data-cite="!GETUSERMEDIA#dfn-constraints"><code>Constraints</code></dfn>,
+        <dfn data-cite="!GETUSERMEDIA#dfn-capabilities"><code>Capabilities</code></dfn>,
+        <dfn data-cite="!GETUSERMEDIA#idl-def-ConstraintSet"><code>ConstraintSet</code></dfn>,
+        and <dfn data-cite="!GETUSERMEDIA#dfn-settings"><code>Settings</code></dfn>, and
```
, and Setting, and types => , Settings and types

anssiko replied:
I prefer Oxford comma: https://en.wikipedia.org/wiki/Serial_comma :-)

astojilj commented on index.html:
```diff
@@ -376,7 +387,8 @@
       </p>
       <p>
-        The meaning of dictionary member being <dfn data-cite="!WEBIDL#dfn-present">present</dfn> or <dfn data-cite="!WEBIDL#dfn-present">not present</dfn> is defined in [[WEBIDL]].
+        The meaning of dictionary member being <dfn data-cite="!WEBIDL#dfn-present">present</dfn> or <dfn data-cite="!WEBIDL#dfn-present">not present</dfn>, and its <dfn data-cite="!WEBIDL#dfn-dictionary-member-default-value">default value</dfn> are
```
"being present or not present, and its default value are"  => "being present or not present and having a default value are"

anssiko replied:
Ditto

astojilj commented on index.html:
```diff
@@ -600,6 +629,16 @@
+        <p>
+          See the <a>types of constrainable properties</a> for the definitions
+          of <dfn><code>ConstrainDOMString</code></dfn>,
+          <dfn><code>ConstrainDouble</code></dfn>, and
```
Nit: ", and" => " and"

anssiko replied:
Oxford comma again :-)

astojilj commented on index.html:
```diff
@@ -560,26 +572,43 @@
           <a>MediaTrackSupportedConstraints</a> dictionary
         </h2>
+        <p>
+          <a>MediaTrackSupportedConstraints</a> dictionary represents the list
```
This sentence is the same as in main spec, but it is a bit difficult to grasp just from the sentence itself.
Propose to just keep the reference to main spec and add here:
"Partial dictionary <a>MediaTrackSupportedConstraints</a> extends the original dictionary defined in [GETUSERMEDIA]. Boolean value true represents applicable MediaStreamTrack constraint.

anssiko replied:
I think it is good to repeat the text from the main spec here for completeness. Added the proposed extra text with some modifications. Also defined "applicable constraint".

astojilj commented on index.html:
```diff
@@ -560,26 +572,43 @@
-              boolean focalLengthY = true;
-              boolean principalPointX = true;
-              boolean principalPointY = true;
+              boolean focalLengthX = false;
```
This defines the default value, but still allows the user agent to override and return true.
Ideally, we could have only one field in Capabilities and Constraints (videoKind) and have all of the fields in Settings, but it is not possible if we stick to definition in main spec:

Capabilities:
Capabilities is a dictionary containing one or more key-value pairs, where each key MUST be a constrainable property, and each value MUST be a subset of the set of values allowed for that property.
Settings:
"There MUST be a single value for each key and the value MUST be a member of the set defined for that property by getCapabilities()."

So, as we cannot have a field in Settings but no fields with the same name in Constraint/Dependencies we can just set default to false.
We cannot change  _(double or DoubleRange)_ focalLengthX; to _double_ focalLengthX in MediaTrackCapabilities (what we discussed yesterday).
Anyway, just adding a note, I have no objection to the change here.

anssiko replied:
OK.

astojilj commented on index.html:
```diff
@@ -560,26 +572,43 @@
           <a>MediaTrackCapabilities</a> dictionary
         </h2>
+        <p>
+          <a>MediaTrackCapabilities</a> dictionary represents the
```
Replace [the original sentence](http://www.w3.org/TR/mediacapture-streams/#idl-def-MediaTrackCapabilities) by:
Partial dictionary MediaTrackCapabilities extends the original MediaTrackCapabilities dictionary defined in [GETUSERMEDIA].

anssiko replied:
Added.

astojilj commented on index.html:
```diff
@@ -600,6 +629,16 @@
           <code>MediaTrackConstraintSet</code> dictionary
         </h2>
+        <p>
+          <a>ConstraintSet</a> dictionary specifies each member's set of
```
<a>MediaTrackConstraintSet</a> partial dictionary extends the original <a>MediaTrackConstraintSet</a>. It specifies each member's set of allowed values.

I would keep the original sentence too here as it is sufficient to explain.

anssiko replied:
Done.

astojilj commented on index.html:
```diff
@@ -620,6 +659,10 @@
           <code>MediaTrackSettings</code> dictionary
         </h2>
+        <p>
+          <a>MediaTrackSettings</a> dictionary represents the <a>Settings</a>
```
<a>MediaTrackSettings</a> partial dictionary extends the original <a>MediaTrackSettings</a> dictionary. It represents the <a>Settings</a> of a <a>MediaStreamTrack</a> object.

anssiko replied:
Done.

astojilj reviewed (commented):
The new layout looks good. Few nits.

anssiko committed:
Address review feedback

anssiko commented:
@astojilj, thanks for the review! I addressed your feedback in https://github.com/w3c/mediacapture-depth/pull/160/commits/5fe64304f7460975c902cd600a619028cca5e774 and will land now. We can iterate further in subsequent PRs if needed.
